.github/workflows/ci.yml

```diff
@@ -54,23 +53,87 @@ jobs:
     - name: Tag docker image
       run: docker tag docker.pkg.github.com/${OWNER}/${DOCKER_CACHE_REPO}/${DOCKER_TAG} ${DOCKER_TAG}
     - name: Check pre-commit
+      if: github.event_name == 'pull_request'
       run: |
         docker run \
-          -e TRAVIS_COMMIT_RANGE \
+          -e COMMIT_RANGE \
           --memory 7500m \
           --memory-swap 7500m \
           "${DOCKER_TAG}" scripts/travisci/check_precommit.sh
+    - name: Run normal tests
+      run: |
+        ci_env="$(bash <(curl -s https://codecov.io/env))" &&
+        docker run \
```

Comment:
Would your life become simpler here if you were able to use the built-in docker support? https://docs.github.com/en/actions/creating-actions/creating-a-docker-container-action

Perhaps this is an issue for another PR however.